applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -394,29 +396,34 @@ class EigensolverStrategy
         SparseSpaceType::Resize(b,SparseSpaceType::Size1(rMassMatrix));
         SparseSpaceType::Set(b,0.0);
 
-        // Generate lhs matrix. the factor 1 is chosen to preserve
-        // SPD property
+        // if the size of the Matrix is the same as the size of the dofset, then also the dirichlet-dofs are in the matrix
+        // i.e. a BlockBuilder is used
+        const bool matrix_contains_dirichlet_dofs = SparseSpaceType::Size1(rMassMatrix) == this->pGetBuilderAndSolver()->GetDofSet().size();
```

Comment:
this is honestly not the cleanest solution, but it is not possible to use `ApplyDirichletConditions` of the B&S, the implementation/requirements are slightly different